### PR TITLE
sync: update sources to meet hal updates

### PR DESCRIPTION
**Pull request by sylvioalves**

### CODE REVIEW

File: components/bootloader_support/src/esp32/bootloader_esp32.c [hunk 1]
```diff
@@ -211,6 +211,13 @@ esp_err_t bootloader_init(void)
     // print 2nd bootloader banner
     bootloader_print_banner();
 
+#ifdef CONFIG_ESP_SIMPLE_BOOT
```
Comment (raffarost) [inline components/bootloader_support/src/esp32/bootloader_esp32.c:214]: I'm guessing you're just keeping in sync with PR #336

File: components/esp_hw_support/include/esp_private/rtc_clk.h [hunk 2]
```diff
@@ -44,6 +44,12 @@ void rtc_clk_bbpll_add_consumer(void);
  */
 void rtc_clk_bbpll_remove_consumer(void);
 
+/**
+ * @brief Workaround for C2, S3, C6, H2. Trigger the calibration of PLL. Should be called when the bootloader doesn't provide a good enough PLL accuracy.
+*/
+void rtc_clk_recalib_bbpll(void);
```
Comment (raffarost) [inline components/esp_hw_support/include/esp_private/rtc_clk.h:50]: will we include this in the startup flow?

File: components/esp_phy/src/phy_override.c [hunk 3]
```diff
@@ -9,8 +9,7 @@
 #include "esp_private/regi2c_ctrl.h"
 #include "esp_private/sar_periph_ctrl.h"
 #include "esp_private/sar_periph_ctrl.h"
-#include "freertos/FreeRTOS.h"
-
+#include <zephyr/kernel.h>
```
Comment (raffarost) [inline components/esp_phy/src/phy_override.c:12]: just a note: we might not be using this file, as per CMakeLists file check

File: components/esp_rom/patches/esp_rom_hp_regi2c_esp32c6.c [hunk 4]
```diff
@@ -0,0 +1,188 @@
+/*
+ * SPDX-FileCopyrightText: 2019-2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "esp_rom_sys.h"
```
Comment (raffarost) [inline components/esp_rom/patches/esp_rom_hp_regi2c_esp32c6.c:6]: just a note to check if we use or not (and should) this file

File: components/esp_system/include/esp_system.h [hunk 5]
```diff
@@ -33,6 +33,11 @@ typedef enum {
     ESP_RST_DEEPSLEEP,  //!< Reset after exiting deep sleep mode
     ESP_RST_BROWNOUT,   //!< Brownout reset (software or hardware)
     ESP_RST_SDIO,       //!< Reset over SDIO
+    ESP_RST_USB,        //!< Reset by USB peripheral
```
Comment (raffarost) [inline components/esp_system/include/esp_system.h:36]: reminder to check if this prompts any improvements or changes in reset handling

File: components/hal/esp32c6/include/hal/uart_ll.h [hunk 6]
```diff
@@ -7,33 +7,27 @@
 // The LL layer for UART register operations.
 // Note that most of the register operations in this layer are non-atomic operations.
 
+
```
Comment (raffarost) [inline components/hal/esp32c6/include/hal/uart_ll.h:10]: looks like this file was rolled back to a previous version
can we keep latest one? function like `uart_ll_is_enabled` is necessary for PM

File: components/soc/esp32c2/include/soc/soc.h [hunk 7]
```diff
@@ -146,9 +147,7 @@
 #define  EFUSE_CLK_FREQ_ROM                          ( 20*1000000)
 #define  CPU_CLK_FREQ_MHZ_BTLD                       (80)           // The cpu clock frequency (in MHz) to set at 2nd stage bootloader system clock configuration
 #define  CPU_CLK_FREQ                                APB_CLK_FREQ
-#define  APB_CLK_FREQ                                ( CONFIG_XTAL_FREQ * 1000000 )
-#define  SCLK_CLK_FREQ                               ( 60*1000000 )
-#define  MODEM_REQUIRED_MIN_APB_CLK_FREQ             ( 80*1000000 )
+#define  APB_CLK_FREQ                                ( 40*1000000 )
```
Comment (raffarost) [inline components/soc/esp32c2/include/soc/soc.h:150]: we likely want to keep `APB_CLK_FREQ` and `SCLK_CLK_FREQ` as before

File: components/soc/esp32c2/include/soc/soc_caps.h [hunk 8]
```diff
@@ -306,6 +312,7 @@
 #define SOC_WIFI_HW_TSF                           (1)    /*!< Support hardware TSF */
 #define SOC_WIFI_FTM_SUPPORT                      (1)    /*!< Support FTM */
 #define SOC_WIFI_SUPPORT_VARIABLE_BEACON_WINDOW   (1)    /*!< Support delta early time for rf phy on/off */
+#define SOC_WIFI_PHY_NEEDS_USB_WORKAROUND         (1)    /*!< SoC has WiFi and USB PHYs interference, needs a workaround */
```
Comment (raffarost) [inline components/soc/esp32c2/include/soc/soc_caps.h:315]: is this right? afaik C2 has no USB and I remember this being a problem when enabling `wifi`

File: components/soc/esp32c6/include/soc/soc_caps.h [hunk 9]
```diff
@@ -169,10 +158,9 @@
 #define SOC_DS_KEY_CHECK_MAX_WAIT_US (1100)
 
 /*-------------------------- GDMA CAPS -------------------------------------*/
-#define SOC_AHB_GDMA_VERSION            1U
-#define SOC_GDMA_NUM_GROUPS_MAX         1U
-#define SOC_GDMA_PAIRS_PER_GROUP_MAX    3
-#define SOC_GDMA_SUPPORT_ETM            1  // Support ETM submodule
+#define SOC_GDMA_GROUPS                 (1U) // Number of GDMA groups
+#define SOC_GDMA_PAIRS_PER_GROUP        (3)  // Number of GDMA pairs in each group
```
Comment (raffarost) [inline components/soc/esp32c6/include/soc/soc_caps.h:162]: this change will require an update (cleanup) on `dma_esp32_gdma.c`

File: components/soc/esp32c6/include/soc/soc_caps.h [hunk 10]
```diff
@@ -466,12 +436,9 @@
 #define SOC_CRYPTO_DPA_PROTECTION_SUPPORTED     1
 
 /*-------------------------- UART CAPS ---------------------------------------*/
-// ESP32-C6 has 3 UARTs (2 HP UART, and 1 LP UART)
-#define SOC_UART_NUM                    (3)
-#define SOC_UART_HP_NUM                 (2)
```
Comment (raffarost) [inline components/soc/esp32c6/include/soc/soc_caps.h:471]: keeping this change is needed for PM

File: components/bootloader_support/src/esp32/bootloader_esp32.c (hunk 1 above)
Comment (sylvioalves) [reply components/bootloader_support/src/esp32/bootloader_esp32.c:214]: Correct. Well, that one was kept in sync with this.

File: components/esp_rom/patches/esp_rom_hp_regi2c_esp32c6.c (hunk 4 above)
Comment (sylvioalves) [reply components/esp_rom/patches/esp_rom_hp_regi2c_esp32c6.c:6]: I think we should use all patches when available.

File: components/esp_phy/src/phy_override.c (hunk 3 above)
Comment (sylvioalves) [reply components/esp_phy/src/phy_override.c:12]: I added in the build system. Will update it as well.

**Review by wmrsouza — APPROVED** [review]

**Review by marekmatej — APPROVED** [review]

**Pull request merged**